File: python/producer.py
import kafka
import sys
import logging

logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.INFO)
brokerPort = "9092"

try:
    producer = kafka.KafkaProducer(
        bootstrap_servers=["127.0.0.1:" + brokerPort], 
        client_id="admin"
    )

    # produce message
    if len(sys.argv) > 1:
        messages = sys.argv[1:]
        for msg in messages:
            partition = 0 if msg[0] < "N" else 1
            logging.info(f"Sending message {msg} to partition {partition}")
            result = producer.send(
                topic="Users",
                value=msg.encode(),
                partition=partition
            )
    # send() do not immediately push the message but rather batch them and wait
    # without flush(), the app will exit before the messages can be pushed
    producer.flush()
    
except Exception as e:
    logging.error(e)

File: js/producer.js
// produce message

const Kafka = require("kafkajs").Kafka
const getIpAdress = require("./host_ip.js");
const ipAddress = getIpAdress();
const brokerPort = "9092"

const msg = process.argv[2];

run();

async function run(){
    try {
        const kafka = new Kafka({
            "clientId": "admin",
            "brokers": [ipAddress.concat(":").concat(brokerPort)]
        })
        const producer = kafka.producer();
        console.log("Connecting...")
        await producer.connect()
        console.log("Connected!")
        
        const partition = msg[0] < "N" ? 0 : 1;
        const result = await producer.send({
            "topic": "Users",
            "messages": [
                {
                    "value": msg,
                    "partition": partition
                }
            ]
        })
        console.log(`Message: ${msg} - sent successfully! ${JSON.stringify(result)}`)
        await producer.disconnect()
    }
    catch(ex) {
        console.log("Error occurred.")
    }
    finally{
        process.exit(0)
    }
}